### DISCUSSION TITLE
RDF calculation results are not normalized when using "around" in atom group selection

### DISCUSSION THREAD

**Discussion by Chengeng-Yang** [MDAnalysis/mdanalysis#4872] [Q&A]
Dear Developers,

I'm posting this because I have a question regarding RDF calculation inconsistency. I realized that the calculation results are not normalized when I used "around" in atom group selection of one group. 

I have a quick example below:

**Case 1**
```python
from MDAnalysis.tests.datafiles import TPR,XTC
from MDAnalysis.analysis import rdf

u1 = mda.Universe(TPR,XTC)
res60 = u1.select_atoms('resid 60')
water60 = u1.select_atoms('resname SOL')
rdf1 = rdf.InterRDF(res60, water60,
                            nbins=75,  # default
                            range=(0.0, 15.0),  # distance in angstroms 
                           )
rdf1.run()

plt.plot(rdf1.results.bins,rdf1.results.rdf)
```

![image](https://github.com/user-attachments/assets/04b15974-a87e-47de-8dd7-ecc6525c118f)

**Case 2**
```python
from MDAnalysis.tests.datafiles import TPR,XTC
from MDAnalysis.analysis import rdf

u1 = mda.Universe(TPR,XTC)
res60 = u1.select_atoms('resid 60')
water60 = u1.select_atoms('byres resname SOL and around 15 group res60',res60 = res60, updating = True)
rdf1 = rdf.InterRDF(res60, water60,
                            nbins=75,  # default
                            range=(0.0, 15.0),  # distance in angstroms
                           )
rdf1.run()

plt.plot(rdf1.results.bins,rdf1.results.rdf)
```

![image](https://github.com/user-attachments/assets/85245e43-6576-4995-9e20-7ad195b55323)

**Question**

Could anyone help explain why there's such a difference in the rdf calculation and the normalization doesn't work? Usually, norm is set to "rdf" as default. So I was confused how the calculation works. I'm using MDAnalysis 2.4.3 and python 3.8.18. Thanks so much and I appreciate any answers.

**Comment by orbeckst** [comment]
Looking at the source code https://github.com/MDAnalysis/mdanalysis/blob/263bbe65047e535fbad981852ad9c7327826f93d/package/MDAnalysis/analysis/rdf.py#L314-L318 I see that the normalization takes the size of the AtomGroups `g1` and `g2` at the last frame. With an updating AtomGroup (as in your second example), you will get a different number than if you had the same number of waters in the group at every step (first example). Thus, I'd say, the `norm="rdf"` is not guaranteed to work with updating groups.

I'd _think_, `norm=None` will give similar numbers — can you check?

You can raise an issue for this problem (if you have the ["Create issue from this discussion"](https://github.com/MDAnalysis/mdanalysis/issues/new?created_from_discussion_number=4872) link then you can use it directly). At a minimum, RDF should fail if `g1` or `g2` are updating and `norm` is not `None`. Ideally, we calculate the normalization correctly.

**Reply by orbeckst** [reply]
@Chengeng-Yang were you able to check if `norm=None` gave the same numbers?

**Reply by Chengeng-Yang** [reply]
> @Chengeng-Yang were you able to check if `norm=None` gave the same numbers?

Hi, I'm sorry for my late reply. It seems like norm=None doesn't give the same numbers as in my case 2. The number is actually much higher. I attached my results below:


from MDAnalysis.tests.datafiles import TPR,XTC
u1 = mda.Universe(TPR,XTC)
res60 = u1.select_atoms('resid 60')
water60 = u1.select_atoms('resname SOL')
rdf2 = rdf.InterRDF(res60, water60,
                            nbins=75,  # default
                            range=(0.0, 15.0),  # distance in angstroms
                            norm = None        
                           )
rdf2.run()

plt.plot(rdf2.results.bins,rdf2.results.rdf)

![image](https://github.com/user-attachments/assets/9eb5582a-a83b-4399-ad28-d5651a317677)
